include/matx/transforms/reduce.h

```diff
@@ -1442,10 +1442,8 @@ void __MATX_INLINE__ reduce(OutType dest, const InType &in, ReduceOp op,
                    cudaStream_t stream = 0, [[maybe_unused]] bool init = true)
 {
   MATX_NVTX_START("", matx::MATX_NVTX_LOG_API)
-
-  constexpr bool use_cub = OutType::Rank() == 0 || (OutType::Rank() == 1 && InType::Rank() == 2);
   // Use CUB implementation if we have a tensor on the RHS and it's not blocked from using CUB
-  if constexpr (!is_matx_no_cub_reduction_v<ReduceOp> && use_cub) {
+  if constexpr (!is_matx_no_cub_reduction_v<ReduceOp>) {
```

Comment:
It doesn't look like `is_matx_no_cub_reduction_v` is used any more either: is it worth deleting the old code?

Reply:
Yes, please delete that. I believe out reductions should not be used anymore with maybe this one as an exception:
https://github.com/NVIDIA/MatX/issues/940